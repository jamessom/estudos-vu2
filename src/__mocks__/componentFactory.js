import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from '@/router/index';

export default function componentFactory(options = {}) {
  return {
    wrapper: () => {
      const { component, componentOptions, withRouter } = options;
      const localvue = createLocalVue();

      if (withRouter) {
        localvue.use(VueRouter);

        return shallowMount(component, {
          localvue,
          ...componentOptions,
        });
      }

      return shallowMount(component, {
        localvue,
        ...componentOptions,
      });
    },
  };
}
