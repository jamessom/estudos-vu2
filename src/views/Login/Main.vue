<template>
  <section class="hero is-link is-fullheight is-bold">
    <div class="hero-body">
      <div class="column is-desktop is-three-fifths is-offset-one-fifth">
        <form @submit.prevent="onSubmit()">
          <input-base inputType='email' :inputPlaceholder="`Email`" v-model="email" />
          <input-base inputType='password' :inputPlaceholder="`Password`" v-model="password" />

          <button class="button is-success">Login</button>
        </form>
      </div>
    </div>
  </section>
</template>

<script>
import InputBase from '@/components/InputBase.vue';

export default {
  name: 'Login',
  data() {
    return {
      email: '',
      password: '',
    };
  },
  components: {
    InputBase,
  },
};
</script>
