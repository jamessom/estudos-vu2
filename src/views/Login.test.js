import componentFactory from '@/__mocks__/componentFactory';
import Login from './Login.vue';

describe('src/views/Login.vue', () => {
  const factory = componentFactory({
    component: Login,
  });

  it('should renders correctly', () => {
    expect(factory.wrapper().element).toMatchSnapshot();
  });
});
