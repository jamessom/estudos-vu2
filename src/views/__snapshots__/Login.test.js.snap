// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/views/Login.vue should renders correctly 1`] = `
<section
  class="section"
>
  <div
    class="columns"
  >
    <div
      class="column is-4 is-offset-4"
    >
      <div
        class="field"
      >
        <p
          class="control has-icons-left has-icons-right"
        >
          <input
            class="input"
            placeholder="Email"
            type="email"
          />
           
          <span
            class="icon is-small is-left"
          >
            <i
              class="fas fa-envelope"
            />
          </span>
           
          <span
            class="icon is-small is-right"
          >
            <i
              class="fas fa-check"
            />
          </span>
        </p>
      </div>
       
      <div
        class="field"
      >
        <p
          class="control has-icons-left"
        >
          <input
            class="input"
            placeholder="Password"
            type="password"
          />
           
          <span
            class="icon is-small is-left"
          >
            <i
              class="fas fa-lock"
            />
          </span>
        </p>
      </div>
       
      <div
        class="field"
      >
        <p
          class="control"
        >
          <button
            class="button is-success"
          >
            Login
          </button>
        </p>
      </div>
    </div>
  </div>
</section>
`;
